rotkehlchen/chain/evm/accounting/structures.py

```diff
@@ -27,11 +45,41 @@ class TxAccountingTreatment(Enum):
     SWAP = 0
 
 
-class TxEventSettings(NamedTuple):
-    """Settings for an event generated by a decoder"""
-    taxable: bool
-    count_entire_amount_spend: bool
-    count_cost_basis_pnl: bool
-    method: Literal['acquisition', 'spend']
-    accounting_treatment: Optional[TxAccountingTreatment] = None
-    accountant_cb: Optional[AccountantCallback] = None
+class HistoryBaseEntrySettings():
+    """Settings for history base entry accounting"""
+    def __init__(
+            self,
+            taxable: bool,
+            count_entire_amount_spend: bool,
+            count_cost_basis_pnl: bool,
+            method: Literal['acquisition', 'spend'],
+            accounting_treatment: Optional[TxAccountingTreatment] = None,
+            accountant_cb: Optional[HistoryBaseEntriesAccountantCallback] = None,
+    ):
+        self.taxable = taxable
+        self.count_entire_amount_spend = count_entire_amount_spend
+        self.count_cost_basis_pnl = count_cost_basis_pnl
+        self.method = method
+        self.accounting_treatment = accounting_treatment
+        self.accountant_cb = accountant_cb
+
+
+class TxEventSettings(HistoryBaseEntrySettings):
+    """Settings for an evm event generated by a decoder"""
+    def __init__(
```

Comment:
Why is this one child of the other? and why do you need two classes? aren't they doing the same?

Reply:
Yes, they are doing the same, I made 2 classes for typing purposes. Accounting settings that are defined in evm modules expect the event and the iterator to be for evm events, so I made a special class for this.